File: subscribers/go/all-entitled-datasets/all-entitled-datasets.go
package main

import (
    "github.com/aws/aws-sdk-go/aws"
    "github.com/aws/aws-sdk-go/aws/session"
    "github.com/aws/aws-sdk-go/service/dataexchange"
    "fmt"
)

func main() {
    // Initialize a session that the SDK will use to load
    // credentials from the shared credentials file ~/.aws/credentials
    // and region from the shared configuration file ~/.aws/config.
    sess := session.Must(session.NewSessionWithOptions(session.Options{
      SharedConfigState: session.SharedConfigEnable,
    }))
    svc := dataexchange.New(sess)

    dataSetList, err := svc.ListDataSets(&dataexchange.ListDataSetsInput{
      Origin: aws.String("ENTITLED"),
    })

    if err != nil {
      fmt.Println(err.Error())
      return
    }

    for _, dataSet := range dataSetList.DataSets {
      fmt.Printf("%#s/%#s: %#s\n %#s\n", *dataSet.OriginDetails.ProductId, *dataSet.Id, *dataSet.Name, *dataSet.Description)
    }
}

File: subscribers/ruby/all-entitled-datasets/all-entitled-datasets.rb
require 'aws-sdk-dataexchange'

Aws.config.update({
  region: 'us-east-1',
  credentials: Aws::Credentials.new(
    ENV['AWS_ACCESS_KEY_ID'], 
    ENV['AWS_SECRET_ACCESS_KEY']
  )
})

dx = Aws::DataExchange::Client.new

dx.list_data_sets(origin: 'ENTITLED').each do |response|
  response.data_sets.each do |data_set|
    puts "#{data_set.origin_details.product_id}/#{data_set.id}: #{data_set.name}\n  #{data_set.description}"
  end
end
